workers/unity/Packages/com.improbable.gdk.tools/LocalLaunch.cs

```diff
@@ -42,17 +54,154 @@ public static void BuildConfig()
+
+                process.Dispose();
+                process = null;
+            };
+        }
+
+        private static string GetClientLogFilename()
+        {
+            var logConfigPath = Path.Combine(SpatialProjectRootDir, "workers", "unity");
+            var configFileJson = File.ReadAllText(Path.Combine(logConfigPath, ClientConfigFilename));
+            var dict = Json.Deserialize(configFileJson);
+            var defaultLogFileName = "*unityclient.log";
+            Dictionary<string, object> tempDict;
+
+            var currentOS = "windows";
+            if (Application.platform == RuntimePlatform.OSXEditor)
+            {
+                currentOS = "macos";
+            }
+
+            if (!dict.TryGetValue("external", out var externalValue)) {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key 'external'.");
+                return defaultLogFileName;
+            }
+
+            tempDict = externalValue as Dictionary<string, object>;
+
+            if (!tempDict.TryGetValue("default", out var defaultValue))
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key 'default' within 'external'.");
+                return defaultLogFileName;
+            }
+
+            tempDict = defaultValue as Dictionary<string, object>;
+
+            if (!tempDict.TryGetValue(currentOS, out var currentOSValue))
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key '{currentOS}' within 'external' -> 'default'.");
+                return defaultLogFileName;
+            }
+
+            tempDict = currentOSValue as Dictionary<string, object>;
+
+            if (!tempDict.TryGetValue("arguments", out var argumentsValue))
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain key 'arguments' within 'external' -> 'default' -> '{currentOS}'.");
+                return defaultLogFileName;
+            }
+
+            var arguments = (IList) argumentsValue;
+
+            int argumentIdx = 0;
+            for (argumentIdx = 0; argumentIdx < arguments.Count; argumentIdx++)
+            {
+                // The next argument would be the name of the log file
+                if (arguments[argumentIdx].Equals("-logfile"))
+                {
+                    argumentIdx++;
+                    break;
+                }
+            }
+
+            // Logger file not found - using default one
+            if (argumentIdx >= arguments.Count)
+            {
+                Debug.LogError($"Config file {ClientConfigFilename} doesn't contain '-logfile' argument within 'external' -> 'default' -> '{currentOS}' -> arguments.");
+                return defaultLogFileName;
+            }
+
+            // Strip any relative pathing
+            var logFileName = ((string)arguments[argumentIdx]).Substring(((string) arguments[argumentIdx]).LastIndexOf('/'));
```

Comment:
What happens here if the user gives a log file path like: 

`../../my-special-logs/client.log`? 

Won't this just return `client.log`?

Reply:
Can use `Path.GetFullPath` if appropriate

Reply:
That's correct - I've based the assumption of retrieving the filename only rather than combining the paths based on the current way the spatial logs are operated on:

    var logPath = Path.Combine(SpatialProjectRootDir, "logs");
     var latestLogFile = Directory.GetFiles(logPath, "spatial_*.log")
                    .Select(f => new FileInfo(f))
                    .OrderBy(f => f.LastWriteTimeUtc).LastOrDefault();

Reply:
Nothing requires the user to place the log in `logs` however though - we should be able to resolve the path regardless of where it points (as its relative to the json file)

Reply:
Fixed now.